### ISSUE TITLE
examples/predict-with-pretrained.. gives different output values for gpu and cpu run.

### ISSUE THREAD

**Issue by erogol** [dmlc/mxnet#808]
I got different values for gpu and cpu call of the model in mxnet/example/notebooks/predict-with-pretrained-model.ipynb

For this image : http://cdn.phys.org/newman/gfx/news/hires/2012/1-bmw.jpg

in gpu mode I got following top 5 values 

 [  1.20295647e-06   3.80782865e-07   1.01220928e-06   6.29524550e-07  3.61772209e-06]

['n04285008 sports car, sport car', 'n03459775 grille, radiator grille', 'n03100240 convertible', 'n02974003 car wheel', 'n04037443 racer, race car, racing car'])

in cpu mode I got these                           

 [  1.57485408e-06   4.32469506e-07   1.05921958e-06   5.58813952e-07  3.77878655e-06]

['n03459775 grille, radiator grille', 'n04285008 sports car, sport car', 'n02974003 car wheel', 'n03100240 convertible', 'n03930630 pickup, pickup truck']

So cpu version is better for this example.

**Comment by piiswrong** [comment]
@tqchen could you have a look?

**Comment by tqchen** [comment]
There could be difference between CPU and GPU, due to the difference in computaion in floating points.

**Comment by erogol** [comment]
but why this is stable in Caffe then. It is so notable in MxNet

**Comment by tqchen** [comment]
In this case, seems most scores are relatively small, which could cause such difference. 
The difference could amplify with nets that comes with batch normalization (which is in the inception-bn we used). I am not sure if this is the exact cause of the problem though. 

If we can get some comparison on, say exact same model on other toolkits, we could get a better sense on what was going on

**Comment by erogol** [comment]
I observe that without cudnn cpu and gpu works concurring. With cudnnV3 divergence is the most and with cudnnV4 results are more similar. However cudnnV3 gives the best speed, 94 samples/sec for 32 batch size of inception net . cudnnV4 82 samples/sec  and without cudnn 57 samples/sec. In the end, speed and reliability has reverse move.

**Comment by zhenzhen1022** [comment]
I just start build MXnet and I have one problem need your help. I predicted using the trained model, that was provided in examples, the model trained done,however the file 'predict-with-pretrained-model.ipynb' has one mistake:../notebooks/predict-with-pretrained-model.ipynb: line 237: syntax error: unexpected end of file.while I used gedit command to check the mistake and I found that there were no line 237. so I don't what wrong here. Did anyone occur the same problem? pls share me ur solution, thx so much.
